eidas_node/tests/proxy_service/test_cznia_views.py

```diff
@@ -0,0 +1,92 @@
+from base64 import b64encode
+from copy import deepcopy
+from typing import BinaryIO, cast
+from unittest.mock import patch
+
+from django.test import RequestFactory, SimpleTestCase, override_settings
+from django.urls import reverse
+
+from eidas_node.models import LightResponse, Status
+from eidas_node.proxy_service.cznia.views import CzNiaResponseView
+from eidas_node.proxy_service.views import IdentityProviderResponseView
+from eidas_node.tests.proxy_service.test_views import DATA_DIR
+from eidas_node.tests.test_models import LIGHT_RESPONSE_DICT
+from eidas_node.tests.test_storage import IgniteMockMixin
+from eidas_node.utils import dump_xml
+
+
+@override_settings(ROOT_URLCONF='eidas_node.proxy_service.cznia.urls')
+class TestCzNiaResponseView(IgniteMockMixin, SimpleTestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.url = reverse('identity-provider-response')
+        # self.addCleanup(self.mock_ignite_cache())
+
+    def test_get_saml_response_fix_not_needed(self):
+        self.maxDiff = None
+
+        for name in 'saml_response_failed.xml', 'saml_response.xml':
+            with self.subTest(name=name):
+                with cast(BinaryIO, (DATA_DIR / name).open('rb')) as f:
+                    saml_response_xml = f.read()
+
+                view = CzNiaResponseView()
+                view.request = self.factory.post(self.url,
+                                                 {'SAMLResponse': b64encode(saml_response_xml).decode('ascii')})
+                saml_response = view.get_saml_response(None)
+                self.assertXMLEqual(dump_xml(saml_response.document).decode('utf-8'),
+                                    saml_response_xml.decode('utf-8'))
+
+    def test_get_saml_response_fix_failure_response(self):
+        self.maxDiff = None
+        with cast(BinaryIO, (DATA_DIR / 'saml_response_failed_nia_fixed.xml').open('rb')) as f:
+            saml_response_xml_fixed = f.read()
+
+        for name in 'saml_response_failed_nia.xml', 'saml_response_failed_nia_decrypted.xml':
+            with self.subTest(name=name):
+                with cast(BinaryIO, (DATA_DIR / name).open('rb')) as f:
+                    saml_response_xml = f.read()
+
+                view = CzNiaResponseView()
+                view.request = self.factory.post(self.url,
+                                                 {'SAMLResponse': b64encode(saml_response_xml).decode('ascii')})
+                saml_response = view.get_saml_response(None)
+                self.assertXMLEqual(dump_xml(saml_response.document).decode('utf-8'),
+                                    saml_response_xml_fixed.decode('utf-8'))
+
+    def test_create_light_response_no_strip_prefix(self):
```

Comment:
This still needs the override in case someone runs the tests with different settings.

Reply:
The override added.